SU2_CFD/src/solvers/CNEMONSSolver.cpp

```diff
@@ -1077,6 +1078,102 @@ void CNEMONSSolver::BC_Smoluchowski_Maxwell(CGeometry *geometry,
   END_SU2_OMP_FOR
 }
 
+void CNEMONSSolver::BC_RadiativeEquilibrium_Wall(CGeometry *geometry,
+                                                 CSolver **solver_container,
+                                                 CNumerics *conv_numerics,
+                                                 CNumerics *sour_numerics,
+                                                 CConfig *config,
+                                                 unsigned short val_marker) {
+
+  /*--- Local variables ---*/
+  const auto Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+  const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+  const su2double emissivity = config->Get_Emissivity(Marker_Tag);
+    
+  /*--- Get the locations of the primitive variables ---*/
+  const unsigned short T_INDEX       = nodes->GetTIndex();
+  const unsigned short TVE_INDEX     = nodes->GetTveIndex();
+  const unsigned short RHO_INDEX     = nodes->GetRhoIndex();
+  const unsigned short RHOCVTR_INDEX = nodes->GetRhoCvtrIndex();
+
+  /*--- Loop over all of the vertices on this boundary marker ---*/
+  SU2_OMP_FOR_DYN(OMP_MIN_SIZE)
+  for(auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+
+    const auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+    /*--- Check if the node belongs to the domain (i.e, not a halo node) ---*/
+    if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+    /*--- Compute dual-grid area and boundary normal ---*/
+    const auto Normal = geometry->vertex[val_marker][iVertex]->GetNormal();
+    su2double Area = GeometryToolbox::Norm(nDim, Normal);
+
+    /*--- Initialize the convective & viscous residuals to zero ---*/
+    for (auto iVar = 0u; iVar < nVar; iVar++) {Res_Visc[iVar] = 0.0;}
```

Comment:
Try not to use these "helper" member variables of the solver `Res_Visc` they make everything worse.
```suggestion
    su2double Res_Visc[MAXNVAR] = {0.0};
```